source/MaterialXGenGlsl/GlslResourceBindingContext.cpp

```diff
@@ -11,15 +11,21 @@ namespace MaterialX
 //
 // GlslResourceBindingContext
 //
-GlslResourceBindingContext::GlslResourceBindingContext()
+GlslResourceBindingContext::GlslResourceBindingContext(
+    size_t uniformBindingLocation, size_t samplerBindingLocation) :
+    _hwInitUniformBindLocation(uniformBindingLocation),
+    _hwInitSamplerBindLocation(samplerBindingLocation)
```

Comment:
You don't initialize _hwUniformBindLocation , and _hwSamplerBindLocation  here.
Guess you should call initialize().

Reply:
It's done when the context is initialized